Sources/edge/Commands/Devices.swift

```diff
@@ -0,0 +1,152 @@
+import ArgumentParser
+import Foundation
+import Logging
+
+protocol DeviceDiscovery {
+    func findUSBDevices(logger: Logger) async -> [USBDevice]
+    func findEthernetInterfaces(logger: Logger) async -> [EthernetInterface]
+}
+
+struct DevicesCommand: AsyncParsableCommand {
+    static let configuration = CommandConfiguration(
+        commandName: "devices",
+        abstract: "List USB and Ethernet devices connected to the system"
+    )
+
+    enum DeviceType: String, ExpressibleByArgument {
+        case usb, ethernet, all
+    }
+
+    enum OutputFormat: String, ExpressibleByArgument {
+        case json, text
+    }
+
+    @Option(help: "Device types to list (usb, ethernet, or both)")
+    var type: DeviceType = .all
+
+    @Flag(name: [.customShort("j"), .long], help: "Output in JSON format")
+    var json: Bool = false
+
+    func listDevices(
+        usbDevices: [USBDevice],
+        ethernetInterfaces: [EthernetInterface],
+        logger: Logger
+    ) async {
+        let format = json ? OutputFormat.json : OutputFormat.text
+
+        // If using JSON format and we have both types of devices,
+        // use the DevicesCollection struct for combined output
+        if format == .json && type == .all {
```

Comment:
I'd prefer using a switch statement for things like this, because that prevents us from forgetting scenarios when we add more formats or interfaces to discover through.